include/boost/spirit/home/qi/detail/expectation_failure.hpp

```diff
@@ -0,0 +1,32 @@
+/*=============================================================================
+Copyright (c) 2001-2011 Joel de Guzman
+
+Distributed under the Boost Software License, Version 1.0. (See accompanying
+file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+=============================================================================*/
+#if !defined(SPIRIT_EXPECTATION_FAILURE_JULY_19_2016)
+#define SPIRIT_EXPECTATION_FAILURE_JULY_19_2016
+
+#if defined(_MSC_VER)
+#pragma once
+#endif
+
+#include <stdexcept>
+
+namespace boost { namespace spirit { namespace qi {
+    template <typename Iterator>
+    struct expectation_failure : std::runtime_error
+    {
+        expectation_failure(Iterator first_, Iterator last_, info const& what)
```

Comment:
The test suite does not build without compilation errors or failed tests with  gcc 6.1.0 (MinGW) either. This applies to the develop branch. I can provide an output file if it is of interest. gcc complains (almost) about the same things my several msvc toolsets do.